### PR TITLE
Multiple fixes for issues found by Cppcheck

### PR DESCRIPTION
**Pull request by Nekto89**
Cppcheck 1.83
Mostly reducing scope of variables.

### CODE REVIEW

**Comment by bagder** [comment]
Remember 'make checksrc':

~~~
./lib506.c:134:2: warning: Trailing whitespace (TRAILINGSPACE)
   
  ^
./lib512.c:39:2: warning: Trailing whitespace (TRAILINGSPACE)
   
  ^
./lib556.c:74:4: warning: Trailing whitespace (TRAILINGSPACE)
     
    ^
./lib586.c:102:2: warning: Trailing whitespace (TRAILINGSPACE)
   
  ^

**Comment by Nekto89** [comment]
I should have thought of something like this 
git ls-files -z -- '*.c' '*.h' | xargs -0 sed -i 's/ \+$//'
instead of manually fixing files from error logs

File: lib/sendf.c [hunk 1]
```diff
@@ -835,12 +835,12 @@ static int showit(struct Curl_easy *data, curl_infotype type,
   return rc;
 }
 
-int Curl_debug(struct Curl_easy *data, curl_infotype type,
-               char *ptr, size_t size,
+int Curl_debug(struct Curl_easy *handle, curl_infotype type,
+               char *data, size_t size,
```
[the remark this answers is not in the document]
Comment (Nekto89) [reply lib/sendf.c]: so should I change .h file instead?

File: tests/libtest/lib1513.c [hunk 2]
```diff
@@ -58,7 +58,7 @@ int test(char *URL)
   easy_setopt(curl, CURLOPT_TIMEOUT, (long)7);
   easy_setopt(curl, CURLOPT_NOSIGNAL, (long)1);
   easy_setopt(curl, CURLOPT_PROGRESSFUNCTION, progressKiller);
-  easy_setopt(curl, CURLOPT_PROGRESSDATA, NULL);
+  easy_setopt(curl, CURLOPT_PROGRESSDATA, (void *)NULL);
```
[the remark this answers is not in the document]
Comment (Nekto89) [reply tests/libtest/lib1513.c]: https://github.com/danmar/cppcheck/blob/7b106c067aa85e56116d24785e78da102bbb4114/lib/checkother.cpp#L2425

**Comment by Nekto89** [comment]
1 test failed. But before last 3 commits it wasn't failing. They shouldn't change behavior at all.

File: tests/libtest/lib1513.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (bagder) [reply tests/libtest/lib1513.c]: Thanks for the reference. So NULL *can* be an integer like 0 or 0L according to ANSI C, but POSIX says `The macro shall expand to an integer constant expression with the value 0 cast to type void *.`

But if we would take this to heart all calls to all `curl_*_setopt()` functions with a NULL need that NULL typecasted and I would find that awefully annoying and ugly. The mere fact that no user ever had a problem with this construct is also a pretty strong indication that there really isn't any modern >32 bit system that uses a NULL that isn't a pointer.

A modern compiler would also warn on our code all over if NULL wasn't a pointer and we've never seen such compiler warnings.

All this taken into account, I think we should leave our NULL pointers non-casted, even in the setopt functions.

File: lib/sendf.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (bagder) [reply lib/sendf.c]: yes I think so, thanks!

**Comment by bagder** [comment]
Correct, that test 1455 has showed up as a failure intermittently lately and is unrelated to your changes.

**Comment by bagder** [comment]
Thanks!

**Pull request closed without merging**